File: biseccion.py
def biseccion(f, a, b, tol):

    if f(a)*f(b) >= 0:
        print("Error: La función no cambia de signo en el intervalo dado.")
        return None
    else:
        c = (a+b)/2
        while abs(f(c)) > tol:
            if f(a)*f(c) < 0:
                b = c
            else:
                a = c
            c = (a+b)/2
        return c

# F(x)
def f(x):
    return x**2 - 2

# intervalo y toleracia
a = 0
b = 2
tol = 1e-6

raiz = biseccion(f, a, b, tol)
print("La raíz de la función es:", raiz)

File: Backend/metodos/MetodosJava/Biseccion.java
import java.util.function.Function;

public class Biseccion {
    public static double biseccion(Function<Double, Double> f, double a, double b, double tol) {
        if (f.apply(a) * f.apply(b) >= 0) {
            System.out.println("Error: La función no cambia de signo en el intervalo dado.");
            return Double.NaN;
        } else {
            double c = (a + b) / 2;
            while (Math.abs(f.apply(c)) > tol) {
                if (f.apply(a) * f.apply(c) < 0) {
                    b = c;
                } else {
                    a = c;
                }
                c = (a + b) / 2;
            }
            return c;
        }
    }
}
